.github/workflows/cd.yml

```diff
@@ -22,28 +22,39 @@ on:
         type: string
         required: False
 
-env:
-  API_VERSION: dev
-  WORKER_VERSION: latest
-  CLI_VERSION: latest
 
 jobs:
   set-component-versions:
     runs-on: ubuntu-latest
+    outputs:
+      api_version: ${{ steps.api_version.outputs.result }}
+      worker_version: ${{ steps.worker_version.outputs.result }}
+      cli_version: ${{ steps.cli_version.outputs.result }}
+
     steps:
-    - name: use custom if workflow_dispatch
+
+    - id: api_version
+      name: dynamic input api version
+      run: |
+        if [[ ${{ inputs.api_version }} == "" ]]; then echo dev; else echo ${{ inputs.api_version }}; fi
```

Comment:
Are you sure you shouldn't `echo "dev"`? 
I ask because you expect `inputs.api_version` to be a string and are we sure that `dev` only won't be interpreted as something else and not a string?

Reply:
I think it is not the latest commit.